### PR TITLE
Table list of semantic descriptors to improve visibility (2)

### LINKED ISSUE #196: ASDドキュメント改善案
### ラベル
- `go`の遷移はラベルを省略する
    - シンプルにするため
- 名前は`go`+ `rtのID`で自動的に決まるため

### 色

 - unsafeとidempotent遷移には色をつける？
 - それとも線をsafeと変える？
 
## 詳細

- 今のページ由来のkey: va lueの形式ではなく、Schema.org用語集のページのように表形式？
- インクリメンタルサーチ？

https://www.app-state-diagram.com/manuals/1.0/ja/schema-org.html

### PR DESCRIPTION
**Pull request by koriym**
* descritorをテーブル形式にして一覧性を優先させたデザインにしました。
* 項目をid, title, contained (descriptors), extra infoと思い切って最初化して一覧性を優先させました。
* Extra infoはMacOS9のティッカーを思わせるレトロでカラフルでありながら統一感のあるものにしました。
* saツールをアップデートして差分確認のためにbaselineで対応しています。

Moved from #197 
Close #196 

---

### Summary of changes

- **Improved documentation clarity**: Updated `DumpDocs.php` by including `ENT_QUOTES` constant for proper HTML entity handling.
- **Enhanced descriptor handling**: Added test cases addressing truncation, inline rendering, and edge cases.
- **Static analysis updates**: Adjusted Psalm and PHPStan baselines to reflect project changes and improve analysis accuracy.
- **Code cleanup**: Removed unused methods and improved code maintainability.

### Checklist

- [ ] Unit tests added/updated
- [ ] Documentation updated
- [ ] Static analysis tools configuration adjusted

## Sourceryによるサマリー

DumpDocsと記述子のドキュメントとテストケース生成を更新し、HTMLレンダリング、コードの保守性、静的解析の改善に焦点を当てています。

バグ修正:
- 記述子の解析とレンダリングに関する潜在的な問題を修正しました。
- 記述子処理におけるエラー処理を改善しました。

機能拡張:
- 長い記述子のテキストの切り捨てを改善しました。
- 記述子メタデータのより堅牢な処理を追加しました。
- 状態遷移のためのUnicodeシンボルのレンダリングを強化しました。
- 記述子のためのより柔軟なHTML生成を実装しました。

CI:
- GitHub Actionsワークフローを更新してPHP 8.4をサポートしました。
- codecovの設定をわずかに寛容になるように変更しました。

ドキュメント:
- セマンティック記述子のMarkdown生成を、より柔軟なHTMLベースのテーブル形式を使用するようにリファクタリングしました。
- セマンティックタグとツールチップを使用した記述子メタデータのスタイリングを改善しました。

テスト:
- 新しいMarkdown生成アプローチに対応するためにテストケースを更新しました。
- 静的解析のベースラインを調整しました。

<details>
<summary>Original summary in English</summary>

## Summary by Sourcery

Update documentation and test case generation for DumpDocs and descriptors, focusing on improving HTML rendering, code maintainability, and static analysis

Bug Fixes:
- Fixed potential issues with descriptor parsing and rendering
- Improved error handling in descriptor processing

Enhancements:
- Improved text truncation for long descriptors
- Added more robust handling of descriptor metadata
- Enhanced Unicode symbol rendering for state transitions
- Implemented more flexible HTML generation for descriptors

CI:
- Updated GitHub Actions workflows to support PHP 8.4
- Modified codecov configuration to be slightly more lenient

Documentation:
- Refactored semantic descriptor markdown generation to use a more flexible HTML-based table format
- Added improved styling for descriptor metadata with semantic tags and tooltips

Tests:
- Updated test cases to accommodate new markdown generation approach
- Adjusted static analysis baselines

</details>

## Summary by CodeRabbit

- **New Features**
  - Added interactive tag filtering by clicking tag links in documentation tables across multiple pages.
  - Introduced richer, visually informative tables for semantic descriptors with type indicators, tooltips, and metadata tags.
  - Enhanced graph edge labels with colored icons and tooltips indicating transition types.
  - Added new scripts to improve static analysis baseline generation and documentation compilation.

- **Bug Fixes**
  - Corrected a typo in a test JSON file for relationship types.

- **Documentation**
  - Translated the LMS ALPS profile from Japanese to English.
  - Consolidated and improved layout and clarity of semantic descriptor documentation tables, legends, and metadata across documentation pages.

- **Style**
  - Improved CSS for tables, tooltips, SVG containers, and metadata tags for better layout, readability, and interactivity.

- **Tests**
  - Added comprehensive tests for semantic descriptor markdown generation and table rendering, including edge cases and coverage scenarios.
  - Updated existing tests to reflect new label and table formats.

- **Chores**
  - Updated static analysis and build tool dependencies to latest versions.
  - Added and integrated static analysis baselines for PHPStan and Psalm.
  - Updated CI workflows to support PHP 8.4 and refined release process.
  - Included additional files in packaging and build scripts.

### CODE REVIEW

File: src/DumpDocs.php [hunk 1]
```diff
@@ -33,96 +41,146 @@
...
-        if ($this->isFragment($value)) {
-            [, $id] = explode('#', $value);
 
-            return " * {$key}: [{$id}](semantic.{$id}.{$this->ext})" . PHP_EOL;
+        // 各プロパティタイプごとに異なるスタイルを適用
+        switch ($key) {
+            case 'def':
+                if ($this->isUrl($value)) {
+                    $displayValue = preg_replace('#^https?://#', '', $value);
+                    if (strlen($displayValue) > 30) {
```
[the remark this answers is not in the document]
Comment (koriym) [reply src/DumpDocs.php:72]: Done!

File: src/DumpDocs.php [hunk 2]
```diff
@@ -33,96 +40,142 @@ final class DumpDocs
...
-### <a id="{$descriptor->id}">{$descriptor->id}</a>
-{$description}{$rt}{$linkRelations}{$descriptorSemantic}
-
-EOT;
+        if (mb_strlen($text) <= $maxLength) {
+            return $text;
+        }
+
+        // 省略する場合は短く表示
+        return mb_substr($text, 0, $maxLength - 3 - 100) . '...';
```
[the remark this answers is not in the document]
Comment (koriym) [reply src/DumpDocs.php:55]: これは意図的なものです。141文字のbaloontipを表示するのに140文字の"省略"を避けるために。

File: docs/bookstore/index.html [hunk 3]
```diff
@@ -306,804 +517,314 @@ <h1>オンライン書店カタログシステム2</h1>
...
-</ul>
-
-</div>
-
 <div class="legend">
-    <div class="legend-item">
+    <div class="legend-item" title="A state element (e.g.  HTML.SPAN, HTML.INPUT, etc.).">
         <span class="legend-icon semantic"></span>
         Semantic
     </div>
```
[the remark this answers is not in the document]
Comment (koriym) [reply docs/bookstore/index.html:524]: htmlファイルは生成されたファイルなので修正対象ではないです。

**Comment by koriym** [comment]
@coderabbitai Approved?

File: src/DumpDocs.php [hunk 4]
```diff
@@ -33,96 +40,138 @@ final class DumpDocs
...
+        switch ($key) {
+            case 'def':
+                if ($this->isUrl($value)) {
+                    $displayValue = preg_replace('#^https?://#', '', $value);
+                    if (mb_strlen($displayValue) > 30) {
+                        $displayValue = mb_substr($displayValue, 0, 27) . '...';
+                    }
+
+                    return sprintf(
+                        '<span class="meta-item"><span class="meta-label">def:</span><span class="meta-tag def-tag"><a href="%s" target="_blank">%s</a></span></span>',
```
Comment (NaokiTsuchiya) [inline src/DumpDocs.php:72]: meta-item や meta-label にマッピングするのはメソッドやオブジェクトに切り出した方がわかりやすいような気がしますが、やり過ぎでしょうか？

Comment (koriym) [reply src/DumpDocs.php:72]: そうします！しかし次のPRでsaのfixをしていて、そこで修正します。
